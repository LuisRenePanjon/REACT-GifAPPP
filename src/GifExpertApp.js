import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () =>{

    const [categories, setCategories] = useState( [ 'Naruto' ]);
    // const handleAdd = () => {
    //     setCategories([...categories, 'Boku no hero']);
    // };


    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory 
                setCategories = { setCategories }
            />
            <hr />
            <ol>
                { 
                    categories.map( category  => (
                        <GifGrid 
                            key={ category }
                            category = { category }
                        />
                    ))
                }
            </ol>
        </>
    );
};

export default GifExpertApp;


