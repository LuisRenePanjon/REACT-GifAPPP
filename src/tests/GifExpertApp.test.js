import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Test in <GifExpertApp />', () => {
    const wrapper = shallow(<GifExpertApp />);
    
    test('should match snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should display category list', () => {
        const categories = ['Naruto', 'Goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe(categories.length); 
    });
    
});
