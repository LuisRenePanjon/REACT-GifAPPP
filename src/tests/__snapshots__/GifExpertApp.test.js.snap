// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifExpertApp /> should display category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`Test in <GifExpertApp /> should match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
